test/cppsuite/tests/burst_inserts.cxx

```diff
@@ -0,0 +1,147 @@
+            }
+            collection &coll;
+            scoped_cursor read_cursor;
+            scoped_cursor write_cursor;
+        };
+
+        /* Collection cursor vector. */
+        std::vector<collection_cursor> ccv;
+        uint64_t collection_count = tc->db.get_collection_count();
+        uint64_t collections_per_thread = collection_count / tc->thread_count;
+        /* Must have unique collections for each thread. */
+        testutil_assert(collection_count % tc->thread_count == 0);
+        for (int i = tc->id * collections_per_thread;
+             i < (tc->id * collections_per_thread) + collections_per_thread && tc->running(); ++i) {
+            collection &coll = tc->db.get_collection(i);
+            ccv.push_back({coll, std::move(tc->session.open_scoped_cursor(coll.name.c_str())),
+              std::move(tc->session.open_scoped_cursor(coll.name.c_str(), "next_random=true"))});
+        }
+
+        uint64_t counter = 0;
+        while (tc->running()) {
+            uint64_t start_key = ccv[counter].coll.get_key_count();
+            uint64_t added_count = 0;
+            bool committed = true;
+            auto &cc = ccv[counter];
+            auto burst_start = std::chrono::system_clock::now();
+            while (tc->running() &&
+              std::chrono::system_clock::now() - burst_start < std::chrono::seconds(90)) {
+                tc->transaction.try_begin();
+                cc.write_cursor->set_key(
+                  cc.write_cursor.get(), tc->key_to_string(start_key + added_count).c_str());
+                cc.write_cursor->search(cc.write_cursor.get());
+
+                /* A return value of true implies the insert was successful. */
+                if (!tc->insert(cc.write_cursor, cc.coll.id, start_key + added_count)) {
+                    tc->transaction.rollback();
+                    added_count = 0;
+                    continue;
+                }
+                added_count++;
+
+                /* Walk our random reader intended to generate cache pressure. */
+                int ret = 0;
+                if ((ret = cc.read_cursor->next(cc.read_cursor.get())) != 0) {
+                    if (ret == WT_NOTFOUND) {
+                        cc.read_cursor->reset(cc.read_cursor.get());
+                    } else if (ret == WT_ROLLBACK) {
+                        tc->transaction.rollback();
+                        added_count = 0;
+                        continue;
+                    } else {
+                        testutil_die(ret, "Unhandled error in cursor->next()");
+                    }
+                }
+
+                if (tc->transaction.can_commit()) {
+                    if (tc->transaction.commit()) {
+                        cc.coll.increase_key_count(added_count);
+                        start_key = cc.coll.get_key_count();
+                    }
+                    added_count = 0;
+                }
+                std::this_thread::sleep_for(std::chrono::milliseconds(10));
```

Comment:
Why is this required? Add a comment to explain

Reply:
It was going too quickly but yeah i'll add a comment.